packages/client/lib/sync/sync.ts

```diff
@@ -76,7 +76,16 @@ export abstract class Synchronizer {
     })
 
     this.config.events.on(Event.CHAIN_UPDATED, async () => {
-      if (this.syncTargetHeight && this.chain.blocks.height.gte(this.syncTargetHeight)) {
+      let targetHeight = this.syncTargetHeight
+      if (!targetHeight) {
+        const best = this.best()
+        if (best?.eth) {
+          targetHeight = new BN(best.eth.status.latestBlock)
+        } else if (best?.les) {
+          targetHeight = new BN(best.les.status.headNum)
+        }
+      }
```

Comment:
I guess we want to have this for directly setting the sync target height on a first iteration.

Have given this a bit more thought. The timestamp from the block is actually also not a good indicator for having the chain synced. For example for a stalled chain with 3 participants and 2 signers where we are the one signer and the other signer dropped some time ago (and we therefore would be on charge to produce a next block) there just won't be any new `NEW_BLOCK_HASHES` triggers that would update the chain and bring us in a synced state so that we could finally decide to start on block producing.

I guess with this combination (setting to the peer height and then soon after to the block height from `NEW_BLOCK_HASHES`) we should already cover a good flavour of scenarios (in the "normal" case there should also somewhat soon be a `NEW_BLOCK_HASHES` message coming in).

One scenario which is not covered yet though: how to decide in a two peer setup where we are the block producer and the other peer is just waiting for us that we "are synced" and therefore should actually start/continue producing blocks? 🤔

Reply:
Have added this in ccc0e09.

Reply:
Ok, I guess this b24dc4a is my first take on this, to wait for some time if we find a best peer with a higher target height (and in the meantime there is also the chance that some `NEW_BLOCK_HASHES` come in) and otherwise set the sync target height to the local DB height and emit a `SYNCHRONIZED` event.

This should now also cover this 2 peer scenario and give us the basics to start on block producing.

(really tricky edge cases here though, I wonder e.g. if we should add an explicit flag to allow block production from genesis onwards to not accidentally start new chains if there is no network connectivity)